src/Microsoft.Sbom.Parsers.Spdx30SbomParser/Parser/SPDX30Parser.cs

```diff
@@ -157,7 +163,299 @@ public SPDX30Parser(
+        switch (typeFromSbom)
+        {
+            case "software_File":
+                entityType = "File";
+                break;
+            case "software_Package":
+                entityType = "Package";
+                break;
+        }
+
+        // If the entity type is in the list of entities that require different NTIA requirements, then add the NTIA prefix.
+        switch (requiredComplianceStandard)
+        {
+            case ComplianceStandard.NTIA:
+                if (this.EntitiesToEnforceComplianceStandardsFor?.Contains(entityType) == true)
+                {
+                    entityType = "NTIA" + entityType;
+                }
+
+                break;
+        }
+
+        var type = assembly.GetType($"Microsoft.Sbom.Parsers.Spdx30SbomParser.Entities.{entityType}") ?? throw new ParserException($"Type \"{typeFromSbom} on {jsonObject} is invalid.");
+
+        return type;
+    }
+
+    public void ValidateNTIARequirements(List<Element> elementsList)
+    {
+        ValidateSbomDocCreationForNTIA(elementsList);
+        ValidateSbomFilesForNTIA(elementsList);
+        ValidateSbomPackagesForNTIA(elementsList);
+    }
+
+    /// <summary>
+    /// Validate that information about the SBOM document is present.
+    /// </summary>
+    /// <param name="elementsList"></param>
+    /// <exception cref="ParserException"></exception>
+    public void ValidateSbomDocCreationForNTIA(List<Element> elementsList)
+    {
+        var spdxDocumentElements = elementsList.Where(element => element is SpdxDocument);
+        if (spdxDocumentElements.Count() != 1)
+        {
+            throw new ParserException("SBOM document is not NTIA compliant because it must only contain one SpdxDocument element.");
+        }
+
+        var spdxDocumentElement = spdxDocumentElements.First();
+
+        var spdxCreationInfoElement = (CreationInfo?)elementsList.
+            Where(element => element.Type == nameof(CreationInfo)).
+            FirstOrDefault(element => ((CreationInfo)element).Id == spdxDocumentElement.CreationInfoDetails)
+            ?? throw new ParserException($"SBOM document is not NTIA compliant because it must have a creationInfo element with ID of {spdxDocumentElement.CreationInfoDetails}");
+    }
+
+    /// <summary>
+    /// Validate that all files have declared and concluded licenses.
+    /// </summary>
+    /// <param name="elementsList"></param>
+    /// <exception cref="ParserException"></exception>
+    public void ValidateSbomFilesForNTIA(List<Element> elementsList)
+    {
+        var fileElements = elementsList.Where(element => element is NTIAFile);
+        foreach (var fileElement in fileElements)
+        {
+            var fileSpdxId = fileElement.SpdxId;
+
+            var fileHasSha256Hash = fileElement.VerifiedUsing.
+                Any(packageVerificationCode => packageVerificationCode.Algorithm ==
+                HashAlgorithm.sha256);
```

Comment:
Maybe remove the line break between lines 366 and 367, so the condition is all on one line?